synapse/lib/agenda.py

```diff
@@ -809,7 +809,17 @@ async def _scheduleLoop(self):
                     logger.warning(mesg,
                                    extra={'synapse': {'iden': appt.iden, 'name': appt.name}})
                 else:
-                    await self._execute(appt)
+                    try:
+                        await self._execute(appt)
+                    except s_exc.SynErr as e:
+                        extra = {'iden': appt.iden, 'name': appt.name, 'user': appt.creator, 'view': appt.view}
+                        user = self.core.auth.user(appt.creator)
+                        if user is not None:
+                            extra['username'] = user.name
+                        logger.exception(f'Agenda error running appointment {appt.iden} {appt.name}: '
+                                         f'{e.get("mesg", str(e))}',
+                                         extra={'synapse': extra})
+                        await self._markfailed(appt, f'error: {e}')
```

Comment:
Should this also have an `except Exception` block to prevent the death of the cron task in the event of an unforseen failure? ( now that we don't need to special case task cancelation... :D )